auto_tutorials_source/tutorial_mc_dropout.py

```diff
@@ -0,0 +1,154 @@
+
+- What Uncertainties Do We Need in Bayesian Deep Learning for Computer Vision? `NeurIPS 2017 <https://browse.arxiv.org/pdf/1703.04977.pdf>`_
+- Dropout as a Bayesian Approximation: Representing Model Uncertainty in Deep Learning `PMLR 2016 <https://browse.arxiv.org/pdf/1506.02142.pdf>`_
+
+Training a ResNet with MC Dropout using TorchUncertainty models and PyTorch Lightning
+-------------------------------------------------------------------------------------
+
+In this part, we train a ResNet with dropout layers, based on the model and routines already implemented in TU.
+
+1. Loading the utilities
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, we have to load the following utilities from TorchUncertainty:
+
+- the cli handler: cli_main and argument parser: init_args
+- the model: resnet18, which lies in the torch_uncertainty.models.resnet module
+- the classification training routine in the torch_uncertainty.training.classification module
+- the datamodule that handles dataloaders: MNISTDataModule, which lies in the torch_uncertainty.datamodule
+- the optimizer wrapper in the torch_uncertainty.optimization_procedures module.
+"""
+
+from torch_uncertainty import cli_main, init_args
+from torch_uncertainty.datamodules import MNISTDataModule
+from torch_uncertainty.baselines import ResNet
+from torch_uncertainty.models.resnet import resnet18
+from torch_uncertainty.routines.classification import ClassificationEnsemble
+from torch_uncertainty.optimization_procedures import optim_cifar10_resnet18
+
+# %%
+# We will also need import the neural network utils withing `torch.nn`.
+#
+# We also import ArgvContext to avoid using the jupyter arguments as cli
+# arguments, and therefore avoid errors.
+
+import os
+from functools import partial
+from pathlib import Path
+
+from torch import nn
+from cli_test_helpers import ArgvContext
+
+# %%
+# 2. Creating the necessary variables
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# In the following, we will need to define the root of the datasets and the
+# logs, and to fake-parse the arguments needed for using the PyTorch Lightning
+# Trainer. We also create the datamodule that handles the MNIST dataset,
+# dataloaders and transforms. Finally, we create the model using the
+# blueprint from torch_uncertainty.models.
+# 
+# It is important to specify the arguments ``version`` as ``mc-dropout``,
+# ``num_estimators`` and the ``dropout_rate`` to use Monte Carlo dropout.
+
+root = Path(os.path.abspath(""))
+
+# We mock the arguments for the trainer
+with ArgvContext(
+    "file.py",
+    "--max_epochs",
+    "1",
+    "--enable_progress_bar",
+    "False",
+    "--version",
+    "mc-dropout",
+    "--dropout_rate",
+    "0.4",
+    "--num_estimators",
+    "11",
+):
+    args = init_args(network=ResNet, datamodule=MNISTDataModule)
+
+net_name = "mc-dropout-resnet18-mnist"
+
+# datamodule
+args.root = str(root / "data")
+dm = MNISTDataModule(**vars(args))
+
+
+model = resnet18(
```

Comment:
We may switch to a lenet to avoid too-long computations as we don't have GitHub runners yet.